ops/lib/__init__.py

```diff
@@ -12,7 +12,12 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-"""Infrastructure for the opslib functionality."""
+"""Infrastructure for the opslib functionality.
+
+DEPRECATED: The ops.lib functionality is deprecated, and is superseded by
+charm libraries (https://juju.is/docs/sdk/library) and regular Python imports.
+We now prefer to do version selection at build (charmcraft pack) time.
```

Comment:
The docs changes are fine, do you want to add a deprecation warning at runtime?

Reply:
Ah, good call -- done.